librz/core/cmd/cmd_analysis.c

```diff
@@ -5894,55 +5899,18 @@ RZ_IPI RzCmdStatus rz_analysis_basic_block_find_paths_handler(RzCore *core, int
 	return status;
 }
 
-typedef enum {
-	CORE_ANALYSIS_SIMPLE, ///< aa
-	CORE_ANALYSIS_DEEP, ///< aaa
-	CORE_ANALYSIS_EXPERIMENTAL, ///< aaaa
-} CoreAnalysisType;
-
-static void core_perform_auto_analysis(RzCore *core, CoreAnalysisType type) {
-	ut64 timeout = rz_config_get_i(core->config, "analysis.timeout");
-	char *debugger = NULL;
-	ut64 old_offset = core->offset;
-	const char *notify = "Analyze all flags starting with sym. and entry0 (aa)";
-	rz_core_notify_begin(core, "%s", notify);
-	rz_cons_break_push(NULL, NULL);
-	rz_cons_break_timeout(timeout);
-	rz_core_analysis_all(core);
-	rz_core_notify_done(core, "%s", notify);
-	rz_core_task_yield(&core->tasks);
-	if (type == CORE_ANALYSIS_SIMPLE || rz_cons_is_breaked()) {
-		goto finish;
-	}
-	// Run pending analysis immediately after analysis
-	// Usefull when running commands with ";" or via rizin -c,-i
-	debugger = core->dbg->cur ? strdup(core->dbg->cur->name) : strdup("esil");
-	if (core->io && core->io->desc && core->io->desc->plugin && !core->io->desc->plugin->isdbg) {
-		// set it only if is debugging
-		RZ_FREE(debugger);
-	}
-	rz_cons_clear_line(1);
-	rz_core_analysis_everything(core, type == CORE_ANALYSIS_EXPERIMENTAL, debugger);
-finish:
-	rz_core_seek(core, old_offset, true);
-	// XXX this shouldnt be called. flags muts be created wheen the function is registered
-	rz_core_analysis_flag_every_function(core);
-	rz_cons_break_pop();
-	RZ_FREE(debugger);
-}
-
 RZ_IPI RzCmdStatus rz_analyze_simple_handler(RzCore *core, int argc, const char **argv) {
-	core_perform_auto_analysis(core, CORE_ANALYSIS_SIMPLE);
+	rz_core_perform_auto_analysis(core, RZ_CORE_ANALYSIS_SIMPLE);
```

Comment:
For this new function - `rz_core_perform_auto_analysis()` I recommend to create a new unit/integration test, that will try all levels of the analysis for some target and compare the results between them - e.g. that some functions are missing on SIMPLE level but present on more advanced. You can see these as an example:
- https://github.com/rizinorg/rizin/blob/dev/test/integration/test_open_analyse_save_load_project.c
- https://github.com/rizinorg/rizin/blob/dev/test/integration/test_analysis_global_var.c

No need to save/load project, just opening file multiple times for every different analysis will suffice.